src/audio/detect_test.c

```diff
@@ -127,6 +127,8 @@ static void notify_host(struct comp_dev *dev)
 	event.num_elems = 0;
 
 	ipc_send_comp_notification(dev, &event);
+	/* Send queued IPC messages right away to wake host up ASAP */
+	ipc_process_msg_queue();
```

Comment:
just for my understanding - how did it work before? Was IPC sending delayed? Until what time?

Reply:
Before the IPC message to host was sent after the draining started and it is fine as long as host is up and running. However when host is in power down state you first need to wake him up - that's why it is essential to send the IPC message first.